qiime/core/type/primitive.py

```diff
@@ -349,6 +349,24 @@ def encode(self, value):
 Float = _Float('Float')
 
 
+class _Bool(_Primitive):
+    _valid_predicates = {Arguments,}
+
+    def _is_element_(self, value):
+        return isinstance(value, bool)
+
+    def decode(self, string):
+        if string not in ('false', 'true'):
+            raise TypeError("%s is neither 'true' or 'false'")
+
+        return string == 'true'
+
+    def encode(self, value):
+        return str(value)
```

Comment:
I think you're reading way too much into the current behavior. Arguably everything you've mentioned is really more of a bug than intentional. I wouldn't consider anything the framework does right now as *canonical*, least of all anything in `qiime.core` and especially not `qiime.core.type` which is probably the most incomplete part of QIIME atm. There also hasn't been any use of `encode` which is why no one has noticed the above yet, so the current implementation is rather generous in it's assumption as to input. I don't know yet if that is necessarily a problem either, but it would be trivially solved by first asking if the value passed to `encode` was `in` the type (that functionality *should* be relatively robust, e.g. `'foo' in Float` -> `False`)